ros1_bridge/__init__.py

```diff
@@ -538,24 +538,27 @@ def determine_common_services(ros1_srvs, ros2_srvs, mapping_rules):
                 break
             for i, ros1_field in enumerate(ros1_fields[direction]):
                 ros1_type = ros1_field[0]
-                ros2_type = str(ros2_fields[direction][i].type)
+                ros2_type = ros2_fields[direction][i].type
                 ros1_name = ros1_field[1]
                 ros2_name = ros2_fields[direction][i].name
-                if ros1_type != ros2_type or ros1_name != ros2_name:
+                if ros1_type != str(ros2_type) or ros1_name != ros2_name:
                     match = False
                     break
+                ros2_cpptype = ros2_type.pkg_name + "::msg::" + ros2_type.type if ros2_type.pkg_name else ""
```

Comment:
This might work for now but for actions the namespace of fields might not be `msg`.

Please use single quotes.

Reply:
No idea what to do about the '::msg::', should it be an argument to the function? (that function seems to be all about services)

Reply:
Actually, thinking about it, isn't this always 'msg' even for actions since it is the type of a field?

Reply:
Even though the field is a message it might still be in the `action` namespace.